app/javascript/dashboard.js

```diff
@@ -0,0 +1,432 @@
+import {getJSON} from './misc';
+import M from 'materialize-css';
+
+function post_garden(name, address, lat, long, contact_name, contact_num, email, image, notes) {
+  const xhr = new XMLHttpRequest();
+  const url = './gardens';
+  xhr.open('POST', url, true);
+  xhr.setRequestHeader('Content-Type', 'application/json');
+  xhr.onreadystatechange = function () {
+    if (xhr.readyState === 4 && xhr.status === 200) {
+      const json = JSON.parse(xhr.responseText);
+    }
+  };
+  const data = JSON.stringify({
+    'name': name,
+    'address': address,
+    'lat' : lat,
+    'long' : long,
+    'contact_name' : contact_name,
+    'contact_number' : contact_num,
+    'email' : email,
+    'image' : image,
+    'notes' : notes
+  });
+  console.log(data);
+  xhr.send(data);
+}
+
+function post_produce(name, duration, image) {
+  const xhr = new XMLHttpRequest();
+  const url = './produces';
```

Comment:
Merge issues with DB restructuring. Renamed table is produce_types I believe (other instances sadly marked... as CR practices are up in the air)